test/contrib/rails/tracer_test.rb

```diff
@@ -19,6 +19,9 @@ class TracerTest < ActionController::TestCase
     assert_equal(Rails.configuration.datadog_trace[:default_service], 'rails-app')
     assert_equal(Rails.configuration.datadog_trace[:template_base_path], 'views/')
     assert Rails.configuration.datadog_trace[:tracer]
+    assert !Rails.configuration.datadog_trace[:debug]
+    assert_equal(Rails.configuration.datadog_trace[:trace_agent_hostname], Datadog::Writer::HOSTNAME)
+    assert_equal(Rails.configuration.datadog_trace[:trace_agent_port], Datadog::Writer::PORT)
```

Comment:
Since we have spotted a problem before, can you add a test that updates these three values so that you can check if users settings are properly changed?

You can use the [update_config][1] since the reset is automatically called during the `teardown`. You can find an example [here][2]. If I'm not wrong, it should work straightforward.

[1]: https://github.com/DataDog/dd-trace-rb/blob/master/test/helper.rb#L148
[2]: https://github.com/DataDog/dd-trace-rb/blob/eb81d45b6e94cc2f7b276f9a0c278e2f076e3ca1/test/contrib/rails/tracer_test.rb#L64